File: Frontend/events-frontend/src/app/reviews/model/review.ts
import {FrontendEvent} from '../../events/model/frontendEvent';
import {FrontendRate} from './Rate';
import {FrontendComment} from '../../comments/model/Comment';

export interface FrontendReview {
  id: number;
  createdAt: Date;
  eventCount: number;
  hidden: Boolean;
  event?: FrontendEvent;
  rate: FrontendRate;
  madeBy: string | null
  comment: FrontendComment | null

}

File: Backend/events/src/main/java/rs/ac/ftn/svt/events/model/entity/Review.java
package rs.ac.ftn.svt.events.model.entity;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "reviews")
public class Review {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column
    private LocalDateTime createdAt;
    @Column
    private Integer eventCount;
    @Column
    private Boolean hidden;
    @OneToOne
    private Event event;
    @OneToOne
    private Rate rate;
    @Column
    private String madeBy;
    @OneToOne
    private Comment comment;

}
